src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,88 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_SIZE = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+
+    private Node<K,V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_SIZE];
+        size = 0;
+    }
 
     @Override
     public void put(K key, V value) {
-
+        grow();
+        int keyHash = getHash(key);
+        if (table[keyHash % table.length] == null) {
+            table[keyHash % table.length] = new Node<>(key, value, keyHash, null);
+            size++;
+        } else {
+            Node<K, V> curNode = table[keyHash % table.length];
+            while (curNode != null) {
+                if (key == null ? curNode.key == null : key.equals(curNode.key)) {
+                    curNode.value = value;
+                    return;
+                }
+                if (curNode.next == null) {
+                    curNode.next = new Node<>(key, value, keyHash, null);
+                    size++;
+                    return;
+                }
+                curNode = curNode.next;
+            }
+        }
     }
 
     @Override
     public V getValue(K key) {
+        Node<K, V> curNode = table[getHash(key) % table.length];
+        while (curNode != null) {
+            if (key == null ? curNode.key == null : key.equals(curNode.key)) {
+                return curNode.value;
+            }
+            curNode = curNode.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int getHash(K key) {
+        int code;
+        return Math.abs(key == null ? 0 : (code = key.hashCode()) ^ (code >>> 16));
+    }
+
+    private void grow() {
+        if (size >= table.length * DEFAULT_LOAD_FACTOR) {
```

Comment:
You should use threshhold for checking whether to grow your table or not (threshold = (int) (DEFAULT_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR);)